### ISSUE TITLE
Backend: filter for loc.venue and filter for category is missing by search in all fields

### ISSUE THREAD

**Issue by GiBill** [Heklaterriol/JEM-Project#91]
models/events
line: 401 and 407

$query->where('(a.title LIKE '.$search.' OR a.alias LIKE '.$search.' OR loc.city LIKE '.$search.' OR loc.state LIKE '.$search.' OR loc.country LIKE '.$search.')');

**Comment by jojo12** [comment]
Did I understand right, the filter in Backend>venues has no specialisations?
The same in Backend>categories?

**Comment by GiBill** [comment]
Backend venues + categories is ok

Backend events for case 7 and default:

						case 7:

							/* search all */
							$query->join('LEFT', '#__jem_cats_event_relations AS rel ON rel.itemid = a.id');
							$query->join('LEFT', '#__jem_categories AS c ON c.id = rel.catid');
							$query->where('(a.title LIKE '.$search.' OR a.alias LIKE '.$search.' OR loc.city LIKE '.$search.' OR loc.state LIKE '.$search.' OR loc.country LIKE '.$search.' OR loc.venue LIKE '.$search.' OR c.catname LIKE '.$search.')');

							break;

						default:

							$query->join('LEFT', '#__jem_cats_event_relations AS rel ON rel.itemid = a.id');
							$query->join('LEFT', '#__jem_categories AS c ON c.id = rel.catid');
							$query->where('(a.title LIKE '.$search.' OR a.alias LIKE '.$search.' OR loc.city LIKE '.$search.' OR loc.state LIKE '.$search.' OR loc.country LIKE '.$search.' OR loc.venue LIKE '.$search.' OR c.catname LIKE '.$search.')');

**Comment by Heklaterriol** [comment]
@GiBill Thanks for you report.

To minimize our workload and before we start to search for it:
- Can you briefly describe the phenomenon. Is it what jojo supposes?
- And is your case 7 / default code the fix or it

**Comment by GiBill** [comment]
jojo12 asked: Did I understand right, the filter in Backend>venues has no specialisations?
The same in Backend>categories?

an my answer is:
Backend venues + categories are oK


Backend events
case 7 and default
This is my suggestion for the definitive solution.

**Comment by jojo12** [comment]
I think, this is an enhancement (to jem3), but it works. But could be done also later.

**Comment by mckillo** [comment]
I agree with GiBill, the case = 7 must search in all other fields (include categories and venues). His change works for this case.
In the case default, I wouldn't touch it. This default case is standard search, It's "search event and location (city, state, country)".

@Heklaterriol @jojo12 See commit  cdefdb0c1883fe75b70037de26e049cfe06b0d00 with solved for case 7 (ALL FIELDS).
